package.json

```diff
@@ -143,6 +143,7 @@
     "electron-store": "^8.0.1",
     "electron-updater": "^5.0.1",
     "filesize": "^8.0.7",
+    "font-list": "^1.4.5",
```

Comment:
needed to get the system's fonts, it says it works in all Windows, Mac and Linux systems https://github.com/oldj/node-font-list

Reply:
I'll quickly chime in that `font-list` has some peculiar behavior on Windows:
* It prefers font names in Chinese
* It will invoke powershell on every call (which will then also be intercepted by defender), so caching them and not constantly invoking the method will be mandatory to everything grinding to a halt
* It genuinely lists all the fonts, which is probably not what you want - a significant amount of fonts only provide glyphs for certain character sets, which makes the list balloon. `Microsoft Tai Le` is not a useful font if you run US English, for example.

Reply:
for 1, what do you mean? like, fonts with chinese characters are listed first? maybe it's a sorting thing?

for 2, I can check if we can cache the list and then add a `refresh` button to refresh the cache

for 3, I don't think I can filter out stuff from the list of fonts, I wouldn't know what to safely remove since I don't know which font somebody would need and which names those fonts could have (what if the font name for some reason matches something we are ignoring?) I'll see if I can change the type of the element to allow the user to type the name + suggestions

Reply:
ok I changed it to cache the list of fonts and added a reload/refresh option, now it should only get the list once and then only when the action is triggered by the user

I now order all fonts using the user's locale, so hopefully it generates a list with a more relevant order of each user

I don't really know how to filter fonts out, I'm worried that a list of things to ignore is hard to maintain and I don't want to have false positives that will hide fonts by mistake

I tried to use a text input + datalist element, but it was impossible to style and acted really glitchy while typing

Reply:
For 1, I mean that it tries to get the name in Chinese, and then in English if it can't get it in Chinese. The names can be localized, but the package only tries those two.

For 3, on Windows you can get a list of applicable fonts based on charset with a GDI function, but I don't know how calling out to C is done in node projects.

Honestly, it's probably not something you can fix without writing your own package, I couldn't find any alternative package that does it properly.